dcos/client.go

```diff
@@ -56,8 +56,8 @@ type service struct {
 }
 
 // NewClient creates a new DC/OS client with default configuration
-func NewClient() (*APIClient, error) {
-	config, err := NewConfigManager(nil).Current()
+func NewClient(opts ...ConfigManagerOpt) (*APIClient, error) {
+	config, err := NewConfigManager(opts...).Current()
```

Comment:
I think this should take dedicated options, with one being eg. `WithConfig(c *Config)`, this would let us having client specific options in the future, (logger, user-agent, etc.).

Reply:
OK I see your point. How about I revert the client changes and we just keep the ConfigManager opts changes in this PR? Can add in client opts changes in future PR.

Reply:
Sounds good